lab-dan/model/storage.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+const objectConstructor = require('./objectConstructor')
+
+let storage = {}
+storage.data = {
+  '1234-test-obj': {
+    id: '1234-test-obj',
+    creationDate: new Date(12/11/2016),
+    name: 'Test',
+    breed: 'Shiba',
+  },
+}
+
+// let fs = blueBird.promisifyAll(require('fs'), { suffix: 'Promised'})
+
+// This is what a monkeyfied fs function looks like
+// fs.readFilePromised('file').then()
+// fs.writeFilePromised('file').then()
+
+storage.fetchItem = function(id) {
+  // return fs.readFilePromised(`${__dirname}/../data/data.json`)
+  //   .then(data => {
+  //     data = JSON.parse(data.toString())
+  //     return data[id]
+  //   })
+  if (!id) {
+    return storage.fetchAll()
+  } else if (typeof storage.data[id] === 'object') {
+    return Promise.resolve({code: 200, data: storage.data[id]})
+  }
+  return Promise.reject('not found\n')
+}
+
+storage.fetchAll = function() {
```

Comment:
I'm not sure if anyone told you otherwise, so won't mark it down, but I've been telling folks that my design decision here would be to use fetchAll to get the IDs, and then read each of the files/objects in storage to return the actual content. 

Again, it's a design choice.